### PR TITLE
Ci workflow

### LINKED ISSUE #1864: [devops] Upgrade the actions-rs/toolchain and remove the rust-toolchain file
Upgrade the actions-rs/toolchain and remove the rust-toolchain file

https://github.com/rooch-network/rooch/blob/c0e8686cbbf464751c2792087dca8e61af9735e0/.github/actions/rust-setup/action.yaml#L11-L15

Screenshot 2024-06-11 at 15 15 15

### PR DESCRIPTION
**Pull request by yourmoonlight**
## Summary

Summary about this PR
1. clean unused workflows
2. use dtolnay/rust-toolchain instead of actions-rs/toolchain, cause it's unmaintained
3. Using relative path to reference the action in the current branch for rust-setup

- Closes #1864

### CODE REVIEW

File: rust-toolchain.toml
Comment (jolestar) [inline rust-toolchain.toml:1]: 这个文件不能删除

File: .github/actions/rust-setup/action.yaml [hunk 1]
```diff
@@ -9,11 +9,9 @@ runs:
     - run: sudo apt-get update && sudo apt-get install build-essential ca-certificates clang curl git libpq-dev libssl-dev pkg-config lsof lld --no-install-recommends --assume-yes
       shell: bash
 
-    - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # pin@v1
-      with:
-        override: true
-        components: rustfmt, clippy
+    - uses: dtolnay/rust-toolchain@1.77.1
```
Comment (jolestar) [inline .github/actions/rust-setup/action.yaml:12]: 这里为啥不用 actions-rs/toolchain 这个了？

Comment (yourmoonlight) [reply .github/actions/rust-setup/action.yaml:12]: > 这里为啥不用 actions-rs/toolchain 这个了？

cause it's unmaintained, [dtolnay/rust-toolchain](https://github.com/dtolnay/rust-toolchain) may be a better choice?

**Comment by yourmoonlight** [comment]
close, completed through other prs.

**Pull request closed without merging**